manifests/init.pp

```diff
@@ -170,41 +170,40 @@
   Boolean $firewall_manage,
 
   # Main smb.conf options
-  Optional[String] $workgroup,
-  Optional[String] $server_string,
-  Optional[String] $netbios_name,
-  Optional[Boolean] $domain_master,
-  Optional[Boolean] $preferred_master,
-  Optional[Boolean] $local_master,
-  Optional[Integer[0, 255]] $os_level,
-  Optional[Boolean] $wins_support,
-  Optional[String] $wins_server,
-  Optional[String] $name_resolve_order,
-  Optional[String] $server_min_protocol,
-  Optional[String] $client_max_protocol,
-  Optional[String] $client_min_protocol,
-  Array[String] $hosts_allow,
-  Array[String] $hosts_deny,
-  Array[String] $interfaces,
-  Optional[Boolean] $bind_interfaces_only,
-  Optional[String] $log_file,
-  Optional[Integer] $max_log_size,
-  Optional[String] $passdb_backend,
-  Optional[Boolean] $domain_logons,
-  Optional[String] $map_to_guest,
-  Optional[String] $security,
-  Optional[Boolean] $encrypt_passwords,
-  Optional[Boolean] $unix_password_sync,
-  Optional[String] $socket_options,
-  Optional[String] $syslog,
-  Variant[Enum['ntlmv1-permitted', 'ntlmv2-only', 'mschapv2-and-ntlmv2-only', 'disabled'], Boolean] $ntlm_auth,
-  Optional[Integer] $machine_password_timeout,
-  Optional[String] $realm,
-  Optional[String] $kerberos_method,
-  Optional[String] $dedicated_keytab_file,
-  Optional[Boolean] $obey_pam_restrictions,
-  Optional[Hash] $idmap_config,
-
+  Variant[Undef, String] $workgroup    =  'WORKGROUP',
```

Comment:
Changed from Optional[String] to Variant[Undef, String] here so that the tests would pass.  This is the accepted way to make a parameter optional and also give it a default value.

Reply:
Oh Did not see this comment earlier. thanks!